rusk-wallet/src/bin/command/history.rs

```diff
@@ -126,14 +126,28 @@ pub(crate) async fn transaction_from_notes(
 
             match ret.iter_mut().find(|th| &th.id == id) {
                 Some(tx) => tx.amount += note_amount,
-                None => ret.push(TransactionHistory {
-                    direction,
-                    height: decoded_note.block_height,
-                    amount: note_amount - inputs_amount,
-                    fee: *gas_spent * tx.gas_price(),
-                    tx: tx.clone(),
-                    id: id.clone(),
-                }),
+                None => {
+                    let fee = *gas_spent * tx.gas_price();
+                    let amount = note_amount - inputs_amount;
+                    let amount = if direction == TransactionDirection::Out {
+                        // The fee should not be included in the amount
+                        if amount > 0.0 {
+                            amount - fee as f64
+                        } else {
+                            amount + fee as f64
```

Comment:
Why you're adding the fee here? I would expect to always be `amount - fee as f64`

Could you explain it a bit more?

Reply:
If the amount is negative, say -200 and the fee is 0.5, the amount will be -200.5 - in this case, to remove the fee from the amount, we add 0.5 to -200.5 to make -200.

Reply:
got what you mean.

But isn't that true also when the amount is greater than 0?
Eg:
I unstake 1000dusk and the fee paid are 0.5; the amount will be 999.5. In this case also we need to add the fee so to have 1000dusk as amount.

I don't get the case when we have to subtract it (IMO never)

Reply:
Yeah, that's right.